sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2286,24 +2304,66 @@ class Analyzer(
         // Holds the resolved generator, if one exists in the project list.
         var resolvedGenerator: Generate = null
 
+        def createGenerate(g: Generator, outer: Boolean, names: Seq[String], child: LogicalPlan) = {
```

Comment:
Shall we put this outside like `private def trimAlias` because this block seems to grow too big?

Reply:
ok